ci/test/test_ci.py

```diff
@@ -1,109 +1,28 @@
-    result = sp.run(wait_cmd, stdout=sp.PIPE, stderr=sp.PIPE, shell=True)
-    if result.returncode != 0:
-        raise Exception(f'hello service was not deployed: {result!r}')
-
-    url = deploy_config.url('hello', f'/sha')
-    sha = sp.check_output(f"curl {url}", shell=True)
-    return sha
-
-
-async def wait_for_redeployment(old_sha):
-    wait_interval = 10
-    elapsed_time = 0
-    while elapsed_time < 300:
-        try:
-            new_sha = wait_for_hello()
-            if new_sha != old_sha:
-                print('hello was redeployed', file=sys.stderr)
-                return
-            elapsed_time += wait_interval
-        except Exception:
-            pass
-        await asyncio.sleep(wait_interval)
-    raise Exception(f'hello service was not redeployed in 300 seconds')
-
-
-@pytest.fixture
-async def gh_client1():
-    session = aiohttp.ClientSession(
-        raise_for_status=True,
-        timeout=aiohttp.ClientTimeout(total=60))
-
-    gh_client = gh_aiohttp.GitHubAPI(session, 'test-ci-1', oauth_token=user1_token)
-    yield gh_client
-    await session.close()
-
-
-@pytest.fixture
-async def gh_client2():
-    session = aiohttp.ClientSession(
-        raise_for_status=True,
-        timeout=aiohttp.ClientTimeout(total=60))
-
-    gh_client = gh_aiohttp.GitHubAPI(session, 'test-ci-2', oauth_token=user2_token)
-    yield gh_client
-    await session.close()
-
 
 async def test_deploy():
-    wait_for_hello()
-
-
-# FIXME: This test requires either putting user1 as an authorized user
-#        or having a fake developer who can authorize the sha
-# async def test_pr_merge(gh_client1, gh_client2):
-#     sha = wait_for_hello()
-#
-#     script = f'''
-# git clone https://{user1_token}@github.com/{org}/{repo}.git
-# cd {repo}
-# git config --global user.email ci@hail.is
-# git config --global user.name ci
-# git checkout -b benign-changes
-# echo "hello" > hello.txt
-# git add hello.txt && git commit -m "add hello.txt"
-# git push --set-upstream origin benign-changes
-# '''
-#     sp.run(script, stdout=sp.PIPE, stderr=sp.PIPE, shell=True)
-#
-#     data = {
-#         'title': 'Benign changes',
-#         'head': 'benign-changes',
-#         'base': 'master'
-#     }
-#
-#     result = await gh_client1.post(f'/repos/{org}/{repo}/pulls',
-#                                    data=data)
-#     pull_number = result['number']
-#
-#     await gh_client2.post(f'/repos/{org}/{repo}/pulls/{pull_number}/reviews',
-#                           data={'event': 'APPROVE'})
-#
-#     await wait_for_redeployment(sha)
+    deploy_config = get_deploy_config()
+    ci_deploy_status_url = deploy_config.url('ci', '/api/v1alpha/deploy_status')
+    headers = service_auth_headers(deploy_config, 'ci')
+    async with aiohttp.ClientSession(
+            raise_for_status=True,
+            timeout=aiohttp.ClientTimeout(total=60)) as session:
+        deploy_state = None
+        while deploy_state is None:
```

Comment:
So there's no timeout anymore since this is run directly on batch2, so this will never fail.  I think we should put in a timeout.

Reply:
None of the CI build steps have timeouts now?

Reply:
also the request_retry_transient_errors could retry forever, no?

Reply:
No.  We don't support timeouts in batch yet, and the timeouts previously came from wait-for and the kubectl wait timeouts against the deployed pod (which is now gone)

Reply:
Yes, forever.